pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java

```diff
@@ -35,14 +37,22 @@
  * <p>If the time column is in tenMinutesSinceEpoch, constructor can be invoked as:
  * <p><code>TimeGranularitySpec(LONG, 10, MINUTES, timeColumnName)</code>
  */
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class TimeGranularitySpec {
   private static final int DEFAULT_TIME_UNIT_SIZE = 1;
-
+  
   private DataType _dataType;
   private TimeUnit _timeType;
   private int _timeUnitSize = DEFAULT_TIME_UNIT_SIZE;
+  /*
+   * Can be either EPOCH (default) or SIMPLE_DATE_FORMAT:format e.g SIMPLE_DATE_FORMAT:yyyyMMdd 
+   */
+  private String _timeFormat = TimeFormat.EPOCH.toString();
   private String _name;
-
+  public enum TimeFormat {
+    EPOCH, //default
+    SIMPLE_DATE_FORMAT 
```

Comment:
Assumption of long time field is made in the controller as well, for retention calculation, segment metadata, etc.